platform/ui/src/contextProviders/ViewportGridProvider.tsx

```diff
@@ -6,17 +6,31 @@ import React, {
   useReducer,
 } from 'react';
 import PropTypes from 'prop-types';
-
+import isEqual from 'lodash.isequal';
 import viewportLabels from '../utils/viewportLabels';
+import getPresentationId from './getPresentationId';
 
 const DEFAULT_STATE = {
-  numRows: null,
-  numCols: null,
-  layoutType: 'grid',
+  activeViewportIndex: 0,
+  cachedLayout: {
+    // byPosition is a map of position to viewport info
+    byPosition: {},
+  },
+  hpInfo: {
+    hangingProtocolId: '',
+    stageId: '',
+    stageIdx: 0,
+  },
```

Comment:
`hpInfo` is fine ...., but then you have `hangingProtocolId` .... , also if stageIdx is index, just add the two extra letters please it is much cleaner. I like the following much more honestly 

```
hangingProtocol: {
  protocolId,
  stageId,
  stageIndex
}
```